### DISCUSSION TITLE
Rector stoping without any message

### DISCUSSION THREAD

**Discussion by CViniciusSDias** [rectorphp/rector#8598] [Q&A]
When running Rector on a big codebase, it starts executing, changes the files correctly, but at some point it simply freezes, without finishing the execution and without showing any type of message.

I wonder if there's anything I can do to fetch more information to understand what is happening.

When I run with --debug, the file names are shown until a point where it also just stops. And it's not always at the same point / file. If I run rector just for the last file it shows in the debug output, it works fine. It seems to be something on big codebases.

I already increased the memory limit but it still happens.

**Comment by CViniciusSDias** [comment]
PS.: When it stops, the resources seem to be freed:

![Screenshot of task manager with a decrease of CPU and RAM usage](https://github.com/rectorphp/rector/assets/6991415/027c2514-e410-48d9-aa18-9d8beabdcf6d)

**Comment by samsonasik** [comment]
You can troubleshoot parallel process https://getrector.com/documentation/troubleshooting-parallel

**Reply by CViniciusSDias** [reply]
> You may get some parallel errors [...]

I don't get any errors. 😅
But I'm running `withoutParallel` to see if something changes. It'll probably take a few hours before I get a result, tough. I'll get back here as soon as I get it.

**Reply by CViniciusSDias** [reply]
Yeah, adding `withoutParallel` does not change it. Its till freezes (both with and without `--debug`)

**Reply by samsonasik** [reply]
You can try to narrow down rules half by half until it found the minimal rule(s) that cause it,
and the expected output.

Read this post: https://tomasvotruba.com/blog/2021/02/01/effective-debug-tricks-narrow-scoping/

**Reply by CViniciusSDias** [reply]
I just came back to this project today. I tested multiple times with different individual rules and it always freezes. It doesn't crash nor display any error messages, as I mentioned before. And it's not always in the same file / percentage.

This is my `rector.php` for the last test:

```
return RectorConfig::configure()
		->withPaths([
			__DIR__ . '/classes',
			__DIR__ . '/conf',
			__DIR__ . '/cronjobs',
			__DIR__ . '/internal',
			__DIR__ . '/packages',
			__DIR__ . '/spec',
			__DIR__ . '/tests',
			__DIR__ . '/tpl',
			__DIR__ . '/utils',
		])
		->withoutParallel()
		->withRules([
			PregReplaceEModifierRector::class,
		]);
```

As I mentioned, any rule I tried gave me the same output. It always freezes close to 51% without crashing nor showing any messages. It just stays like this:
![Screenshot of the frozen command](https://github.com/rectorphp/rector/assets/6991415/82049476-3fb4-4c14-8134-d5d343a09b9c)

**Reply by samsonasik** [reply]
You may can run per-directory to see which directory that cause it, narrow it more to get which file/process that cause it, eg:

```bash
vendor/bin/rector process classess

 vendor/bin/rector process conf

# etc etc 
```
